Assets/Scripts/Models/Area/RoomBehavior.cs

```diff
@@ -387,16 +387,15 @@ public RoomBehavior Clone()
         public void Control(Room room)
         {
             this.Room = room;
-            List<Tile> innerTiles = room.GetInnerTiles();
-            List<Tile> borderTiles = room.GetBorderingTiles();
-
-            List<Tile> allTiles = innerTiles.Union(borderTiles).ToList();
+            HashSet<Tile> innerTiles = room.GetInnerTiles();
+            HashSet<Tile> borderTiles = room.GetBoundaryTiles();
+            innerTiles.UnionWith(borderTiles);
```

Comment:
Yeh true, I'll fix that.  This does actually cut down on a single list building as the old one used to create a new list of the union of the other two.  I'll just make it so its a single liner (well two, one for the `HashSet<Tile> allTiles = room.GetInnerTiles()` and the other for the unionwith).